OpenTabletDriverPlugins/WindowsInk.cs

```diff
@@ -0,0 +1,274 @@
+                pos = filter.Filter(pos);
+
+            // Normalize (ratio of 1)
+            pos.X /= TabletProperties.MaxX;
+            pos.Y /= TabletProperties.MaxY;
+
+            // Scale to tablet dimensions (mm)
+            pos.X *= TabletProperties.Width;
+            pos.Y *= TabletProperties.Height;
+
+            // Adjust area to set origin to 0,0
+            pos -= Input.Position;
+
+            // Rotation
+            if (Input.Rotation != 0f)
+            {
+                var tempCopy = new Point(pos.X, pos.Y);
+                pos.X = (tempCopy.X * _rotationMatrix[0]) + (tempCopy.Y * _rotationMatrix[1]);
+                pos.Y = (tempCopy.X * _rotationMatrix[2]) + (tempCopy.Y * _rotationMatrix[3]);
+            }
+
+            // Move area back
+            pos.X += _halfTabletWidth;
+            pos.Y += _halfTabletHeight;
+
+            // Scale to tablet area (ratio of 1)
+            pos.X /= Input.Width;
+            pos.Y /= Input.Height;
+
+            // Scale to display area
+            pos.X *= Output.Width;
+            pos.Y *= Output.Height;
+
+            // Adjust display offset by center
+            pos.X += Output.Position.X - _halfDisplayWidth;
+            pos.Y += Output.Position.Y - _halfDisplayHeight;
+
+            // Clipping to display bounds
+            if (AreaClipping)
+            {
+                if (pos.X < _minX)
+                    pos.X = _minX;
+                if (pos.X > _maxX)
+                    pos.X = _maxX;
+                if (pos.Y < _minY)
+                    pos.Y = _minY;
+                if (pos.Y > _maxY)
+                    pos.Y = _maxY;
+            }
+
+            // Post Filter
+            foreach (IFilter filter in _postFilters)
+                pos = filter.Filter(pos);
+
+            // Setting cursor position
+            //Create virtual report
+
+            double offsetX = -(32767.0 / Output.Width);
+            double offsetY = -(32767.0 / Output.Height);
+
+            var pos_x = Math.Round(pos.X / _screenArea.Width  * 32767.0 + offsetX); // need to get display size
+            var pos_y = Math.Round(pos.Y / _screenArea.Height * 32767.0 + offsetY);
+            double normpressure = (double)report.Pressure / (double)TabletProperties.MaxPressure;
+            double pressure = Math.Round(normpressure * 8191.0);
+            var TipState = report.Pressure >= 1 ? 0x21 : 0x20;
+            //var TipState = report.Raw[1] & ~0x01;
+            //if (pressure != 0)
+            //{
+            //    TipState |= 1;
+            //}
+
+            VirtualReport.vmultiId = 0x09;
+            VirtualReport.reportLength = Convert.ToByte((int)10);
+            VirtualReport.reportId = Convert.ToByte((int)2);
+            VirtualReport.buttons = Convert.ToByte(TipState);
+            VirtualReport.x = (ushort)pos_x;
+            VirtualReport.y = (ushort)pos_y;
+            VirtualReport.pressure = (ushort)pressure;
```

Comment:
All references to `VirtualReport` are only in this method, this could instead just be a local variable where all of the properties are initialized with the object itself
```cs
var virtualReport = new VirtualReport
{
   // the properties
}
```